src/rootfs/files/scripts/install_nodejs.sh

```
@@ -0,0 +1,37 @@
+# chroot functions
+source "${BUILDROOT}/files/scripts/chroot.sh";
+
+function install_nodejs() {
+    log_info "adding NodeSource repository";
+    chroot "${OUTPUTDIR}" /bin/bash -c 'curl -sL https://deb.nodesource.com/setup_22.x | bash -';
+
+    log_info "installing Node.js";
+    chroot "${OUTPUTDIR}" /bin/bash -c 'DEBIAN_FRONTEND=noninteractive apt install -y nodejs';
+
+    # Verify installation
+    local NODE_VERSION=$(chroot "${OUTPUTDIR}" /bin/bash -c 'node --version' 2>/dev/null || true);
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)